app/containers/Zapper/saga.js

```diff
@@ -41,22 +58,86 @@ function* initializeZapper() {
     );
 
     yield put(
-      zapperDataLoaded({ tokens, vaults, balances: balances[account] }),
+      zapperDataLoaded({
+        tokens,
+        vaults,
+        balances: balances[account],
+        pickleVaults,
+      }),
     );
   } catch (err) {
     console.log(err);
   }
 }
 
+function* migratePickleGauge(action) {
+  const {
+    pickleDepositAmount,
+    zapPickleMigrateContract,
+    tokenContract,
+  } = action.payload;
+  const account = yield select(selectAccount());
+
+  // https://api.zapper.fi/v1/vault-stats/pickle?api_key=5d1237c2-3840-4733-8e92-c5a58fe81b88
+  let lpyveCRVVaultv2 = {};
+  let lpyveCRVDAO = {};
+  try {
+    // yield call(oldPickleGaugeContract.methods.exit().send, { from: account });
+    yield call(
+      tokenContract.methods.approve(
+        zapPickleMigrateContract._address,
+        MAX_UINT256,
+      ).send,
+      { from: account },
+    );
+
+    const picklePrices = yield call(
+      request,
+      getZapperApi('/vault-stats/pickle', {}),
+    );
+    //    incomingLP={quantity of pSUSHI ETH / yveCRV-DAO tokens sent by user}
+    // minPTokens={(quantity of pSUSHI ETH / yveCRV-DAO tokens sent by user * pSUSHI ETH / yveCRV-DAO pricePerToken) /
+    //  pSUSHI yveCRV Vault (v2) / ETH pricePerToken}
+    picklePrices.map((pp) => {
```

Comment:
Make sure that you're checking for prices of the right tokens. 

yveCRV-ETH SLP pJar = 0x5Eff6d166D66BacBC1BF52E2C54dD391AE6b1f48 = vecrvAddress
yvBOOST-ETH SLP pJar = 0xCeD67a187b923F0E5ebcc77C7f2F7da20099e378 = boostAddress

So, it would be minPTokens = ( incomingLP * price. vecrvAddress ) / price.boostAddress